### PR TITLE
feat: anoncreds receive and store

### PR DESCRIPTION
**Pull request by cristianIOHK**
Add support to receive and store anoncreds.
Missing http request to backend, the rust wrapper and unit tests.

### CODE REVIEW

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt [hunk 1]
```diff
@@ -73,8 +73,8 @@ data class RequestCredential @JvmOverloads constructor(
             return RequestCredential(
                 body = Body(
                     goalCode = offer.body.goalCode,
-                    comment = offer.body.comment,
-                    formats = offer.body.formats
+                    comment = offer.body.comment
+//                    formats = offer.body.formats
```
Comment (goncalo-frade-iohk) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt:77]: Delete comment

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt [hunk 2]
```diff
@@ -87,8 +87,8 @@ data class RequestCredential @JvmOverloads constructor(
     @Serializable
     data class Body @JvmOverloads constructor(
         val goalCode: String? = null,
-        val comment: String? = null,
-        val formats: Array<CredentialFormat>
+        val comment: String? = null
+//        val formats: Array<CredentialFormat>
```
Comment (goncalo-frade-iohk) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt:91]: Delete comment

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt [hunk 3]
```diff
@@ -156,7 +156,7 @@ inline fun <reified T : Serializable> RequestCredential.Companion.build(
     }
     return RequestCredential(
         body = RequestCredential.Body(
-            formats = aux.map { it.first }.toTypedArray()
+//            formats = aux.map { it.first }.toTypedArray()
```
Comment (goncalo-frade-iohk) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/RequestCredential.kt:159]: Delete comment

File: atala-prism-sdk/src/commonTest/kotlin/io/iohk/atala/prism/walletsdk/prismagent/PrismAgentTests.kt [hunk 4]
```diff
@@ -391,4 +407,59 @@ class PrismAgentTests {
...
+            attachments = arrayOf(attachmentDescriptor),
+            thid = "1",
+            from = fromDID,
+            to = toDID
+        )
+
+        val invitation = agent.prepareRequestCredentialWithIssuer(did = toDID, offer = offerCredential)
+//        val message = json.decodeFromString<Message>(invitationString)
+        println("stop")
+    }
```
Comment (goncalo-frade-iohk) [inline atala-prism-sdk/src/commonTest/kotlin/io/iohk/atala/prism/walletsdk/prismagent/PrismAgentTests.kt:464]: This test is not correct or is it?

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/PolluxImpl.kt [hunk 5]
```diff
@@ -62,12 +132,14 @@ class PolluxImpl(val castor: Castor) : Pollux {
...
     @Throws(PolluxError.NoDomainOrChallengeFound::class)
-    override fun createRequestCredentialJWT(
+    override fun processCredentialRequestJWT(
         subjectDID: DID,
         privateKey: PrivateKey,
         offerJson: JsonObject
     ): String {
-        val parsedPrivateKey = parsePrivateKey(privateKey)
+        val parsedPrivateKey =
+
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/PolluxImpl.kt:141]: Fixed

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/PolluxImpl.kt [hunk 6]
```diff
@@ -94,6 +166,99 @@ class PolluxImpl(val castor: Castor) : Pollux {
...
+            CredentialType.W3C -> {
+                val w3c: W3CCredential = credential as W3CCredential
+                w3c.toStorableCredential()
+            }
+
+            CredentialType.ANONCREDS_ISSUE -> {
+                val anon: AnonCredential = credential as AnonCredential
+                anon.toStorableCredential()
+            }
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/PolluxImpl.kt:186]: Fixed

File: atala-prism-sdk/src/commonTest/kotlin/io/iohk/atala/prism/walletsdk/prismagent/PrismAgentTests.kt (hunk 4 above)
Comment (cristianIOHK) [reply atala-prism-sdk/src/commonTest/kotlin/io/iohk/atala/prism/walletsdk/prismagent/PrismAgentTests.kt]: Finished this and added another one to test issued credential.

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/mercury/MercuryImpl.kt [hunk 7]
```diff
@@ -137,7 +137,7 @@ class MercuryImpl(
             }
         }
         logger.debug(
-            message = "Sending message with type ${message.piuri}",
+            message = "Sending message with type` ${message.piuri}",
```
Comment (hamada147) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/mercury/MercuryImpl.kt:140]:
```suggestion
            message = "Sending message with type ${message.piuri}",
```

File: atala-prism-sdk/src/commonMain/sqldelight/io.iohk.atala.prism.walletsdk.pluto/data/CredentialMetadata.sq [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+);
+
+insert:
+INSERT INTO CredentialMetadata(id, nonce, linkSecretName, linkSecretBlindingData)
+VALUES (:id, :nonce, :linkSecretName, :linkSecretBlindingData);
+
+fetchCredentialMetadata:
+SELECT *
+FROM CredentialMetadata
+WHERE linkSecretName = ?;
```
Comment (hamada147) [inline atala-prism-sdk/src/commonMain/sqldelight/io.iohk.atala.prism.walletsdk.pluto/data/CredentialMetadata.sq:15]: Missing new line at end of file
```suggestion
WHERE linkSecretName = ?;

```

**Review by hamada147 — APPROVED** [review]

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/OfferCredential.kt [hunk 9]
```diff
@@ -48,12 +48,13 @@ data class OfferCredential @JvmOverloads constructor(
 
         @JvmStatic
         fun makeOfferFromProposedCredential(proposed: ProposeCredential): OfferCredential {
+            println("OfferCredential::makeOfferFromProposedCredential: ${proposed.thid}")
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/issueCredential/OfferCredential.kt:51]: Removed

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/models/AnoncredPayload.kt [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    @SerialName("schema_id")
+    val schemaId: String,
+    @SerialName("cred_def_id")
+    val credDefId: String,
+    @SerialName("key_correctness_proof")
+    val keyCorrectnessProof: KeyCorrectnessProof,
+    val nonce: String,
+    @SerialName("method_name")
+    val methodName: String
+)
```
[the remark this answers is not in the document]
Comment (cristianIOHK) [reply atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/pollux/models/AnoncredPayload.kt:15]: Nop, I will delete it

File: atala-prism-sdk/build.gradle.kts [hunk 11]
```diff
@@ -131,7 +99,9 @@ kotlin {
 
                 api("org.lighthousegames:logging:1.1.2")
 
+                implementation("io.iohk.atala.prism.anoncredskmp:anoncreds-kmp:1.0.1-alpha")
```
Comment (hamada147) [inline atala-prism-sdk/build.gradle.kts:102]:
```suggestion
                implementation("io.iohk.atala.prism.anoncredskmp:anoncreds-kmp:1.0.0")
```

File: .github/workflows/pull-request.yml [hunk 12]
```diff
@@ -24,7 +24,7 @@ on:
 jobs:
 
   build:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```
Comment (cristianIOHK) [inline .github/workflows/pull-request.yml:27]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by goncalo-frade-iohk — APPROVED** [review]

**Pull request merged**